packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -216,15 +217,28 @@ function cachingDisplayProcessor(disp: DisplayProcessor, maxCacheSize = 2500): D
 
   return (value: any) => {
     let v = cache.get(value);
+
     if (!v) {
       // Don't grow too big
       if (cache.size === maxCacheSize) {
         cache.clear();
       }
 
       v = disp(value);
+
+      // re-parse into hex string :(
+      if (v.color && v.color[0] !== '#') {
+        let color = tinycolor(v.color);
+
+        v = {
+          ...v,
+          color: color.getAlpha() < 1 ? color.toHex8String() : color.toHexString(),
+        };
+      }
```

Comment:
This feels a bit fragile, or somehow in the wrong place (cached display processor), is there anywhere else this could go? can we move this to the alpha function instead?

Reply:
the problem is that this op is expensive. ideally threshold.color strings are all in #rgb or #rgba strings so this whole thing can go away, but right now they're either #rgb when chosen from the palette or rgba() when chosen from the Custom tab in the picker.

can we make this uniform?

Reply:
Maybe but would require translating it back to rgba in color picker as that is much easier to edit manually for users (input field in color spectrum picker)

Reply:
agreed, but that's much cheaper to do there than to do it for every datapoint on every update.

if that seems easy/doable, feel free to push the changes to this PR and drop the section above. (it's a hack for sure).

Reply:
I'm not sure where/how to think about existing saved colors... but perhaps we should save colors in a prefered format here:
https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ColorPicker/ColorInput.tsx#L56

Reply:
i changed it to this, but it never reaches this code (this might be a legacy picker?)

```js
    if (newColor.isValid()) {
      let hexColor = newColor.toString(newColor.getAlpha() === 1 ? 'hex6' : 'hex8');
      this.updateColor(hexColor);
    }
```

Reply:
This is used and it reaches this point for me. But this is just the text input at the bottom of the spectrum picker

Reply:
If you are looking for a single place to update stored color it would be: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ColorPicker/ColorPickerPopover.tsx#L56 

My main problem with this would be we would have to translate back to rgba as that is what the user will see and type in the input

Reply:
> My main problem with this would be we would have to translate back to rgba as that is what the user will see and type in the input

doing this conversion in the picker when necessary during config time seems more sensible to me than having to potentially parse `rgb()` strings to determine opacity for each of potentially tens of thousands of data points?

Reply:
Agree fully, just we have to convert back so user can edit in a more friendly format

Reply:
there is ongiong work in https://github.com/grafana/grafana/pull/37173 to improve this, but let's merge this for the fix and improve in a followup. eventually this whole cache should probably be moved into StreamingDataFrame, now that https://github.com/grafana/grafana/pull/35465 is merged and we have more info on how to be smarter about evictions.